linkcheck/url.py

```diff
@@ -446,13 +448,11 @@ def url_split (url):
     hostname is always lowercased.
     Precondition: url is syntactically correct URI (eg has no whitespace)
     """
-    scheme, netloc = urllib.parse.splittype(url)
-    host, document = urllib.parse.splithost(netloc)
-    port = default_ports.get(scheme, 0)
-    if host:
-        host = host.lower()
-        host, port = splitport(host, port=port)
-    return scheme, host, port, document
+    o = urllib.parse.urlparse(url)
+    port = o.port
+    if port is None:
+        port = default_ports.get(o.scheme, 0)
+    return o.scheme, o.hostname, port, o.path
```

Comment:
This seems pretty straightforward.

It might raise ValueError where the old splitport code didn't, so that might have consequences?

Reply:
From the get()? I thought that was pretty robust. What would you have to pass to it to cause an exception?

Reply:
No, it's the urllib.parse.urlparse()'s return object's `port` property that calls `int()` without cathing ValueError:

```
>>> urllib.parse.urlparse('//host:port')
ParseResult(scheme='', netloc='host:port', path='', params='', query='', fragment='')

>>> urllib.parse.urlparse('//host:port').port
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.8/urllib/parse.py", line 174, in port
    raise ValueError(message) from None
ValueError: Port could not be cast to integer value as 'port'
```

The old splitport() helper used the is_numeric_port() helper which checked if the port was all-numeric before passing it to `int()`, so it would've returned the default value.

I like that we're not silently ignoring errors, but I'm not sure the ValueError will be caught and reported as a problem with the URL, rather than cause a traceback and "oops you found a bug" message.